packages/block-editor/src/components/inserter/menu.js

```diff
@@ -478,28 +478,14 @@ export default compose(
 		};
 	} ),
 	withDispatch( ( dispatch, ownProps, { select } ) => {
-		const {
-			showInsertionPoint,
-			hideInsertionPoint,
-		} = dispatch( 'core/block-editor' );
+		const { clientId, destinationRootClientId, isAppender } = ownProps;
 
-		// This should be an external action provided in the editor settings.
-		const {
-			__experimentalFetchReusableBlocks: fetchReusableBlocks,
-		} = dispatch( 'core/editor' );
-
-		// To avoid duplication, getInsertionIndex is extracted and used in two event handlers
-		// This breaks the withDispatch not containing any logic rule.
-		// Since it's a function only called when the event handlers are called,
-		// it's fine to extract it.
-		// eslint-disable-next-line no-restricted-syntax
-		function getInsertionIndex() {
+		const getInsertionIndex = () => {
```

Comment:
Thanks for looking (in https://github.com/WordPress/gutenberg/pull/16708#discussion_r335864484) at the rule in question. (For other readers, it's [this one](https://github.com/WordPress/gutenberg/blob/ed24c2f8946c0b0b2fd7783601697b0b5f6502ae/.eslintrc.js#L91-L94).)

In this case, though, I'd rather keep the exception and its comment as is — IMO it's more explicit, whereas turning the function declaration into a `const` feels a little bit like we're tricking ESLint. :)

Reply:
done